test/releasers/ruby-yoshi.ts

```diff
@@ -0,0 +1,101 @@
+// limitations under the License.
+
+import * as assert from 'assert';
+import {describe, it, afterEach} from 'mocha';
+import {GitHub} from '../../src/github';
+import * as sinon from 'sinon';
+import {stubFilesFromFixtures} from './utils';
+import {buildMockCommit, stubSuggesterWithSnapshot} from '../helpers';
+import {RubyYoshi} from '../../src/releasers/ruby-yoshi';
+
+const sandbox = sinon.createSandbox();
+
+function stubFilesToUpdate(github: GitHub, files: string[]) {
+  stubFilesFromFixtures({
+    fixturePath: './test/updaters/fixtures',
+    sandbox,
+    github,
+    files,
+  });
+}
+
+const TAG_SHA = 'da6e52d956c1e35d19e75e0f2fdba439739ba364';
+
+const COMMITS = [
+  buildMockCommit(
+    'fix(deps): update dependency com.google.cloud:google-cloud-storage to v1.120.0'
+  ),
+  buildMockCommit(
+    'fix(deps): update dependency com.google.cloud:google-cloud-spanner to v1.50.0'
+  ),
+  buildMockCommit('chore: update common templates'),
+];
+
+function stubGithub(releasePR: RubyYoshi) {
+  sandbox
+    .stub(releasePR.gh, 'findMergedReleasePR')
+    .returns(Promise.resolve(undefined));
+  sandbox.stub(releasePR.gh, 'findOpenReleasePRs').returns(Promise.resolve([]));
+  sandbox.stub(releasePR.gh, 'openPR').resolves(22);
+  sandbox.stub(releasePR.gh, 'addLabels');
+  sandbox.stub(releasePR.gh, 'getDefaultBranch').resolves('master');
+}
+
+describe('RubyYoshi', () => {
+  afterEach(() => {
+    sandbox.restore();
+  });
+  const pkgName = 'google-cloud-automl';
+
+  describe('run', () => {
+    it('creates a release PR with a previous release', async function () {
+      const releasePR = new RubyYoshi({
+        github: new GitHub({owner: 'googleapis', repo: 'ruby-test-repo'}),
+        versionFile: 'version.rb',
+        bumpMinorPreMajor: true,
+        monorepoTags: true,
+        packageName: pkgName,
+        lastPackageVersion: '0.5.0',
+      });
+
+      stubSuggesterWithSnapshot(sandbox, this.test!.fullTitle());
+      stubGithub(releasePR);
+      sandbox.stub(releasePR.gh, 'getTagSha').resolves(TAG_SHA);
+      sandbox.stub(releasePR.gh, 'commitsSinceSha').resolves(COMMITS);
+      stubFilesToUpdate(releasePR.gh, ['version.rb']);
+
+      const pr = await releasePR.run();
+      assert.strictEqual(pr, 22);
+    });
+
+    it('creates a release PR with no previous release', async function () {
+      const releasePR = new RubyYoshi({
+        github: new GitHub({owner: 'googleapis', repo: 'ruby-test-repo'}),
+        versionFile: 'version.rb',
+        bumpMinorPreMajor: true,
+        monorepoTags: true,
+        packageName: pkgName,
+      });
+
+      stubSuggesterWithSnapshot(sandbox, this.test!.fullTitle());
+      stubGithub(releasePR);
+      sandbox.stub(releasePR.gh, 'commitsSinceSha').resolves(COMMITS);
+      stubFilesToUpdate(releasePR.gh, ['version.rb']);
+
+      const pr = await releasePR.run();
+      assert.strictEqual(pr, 22);
+    });
+  });
```

Comment:
We probably want an assertion on the proposed version number - testing that the initial version is passed through?

Reply:
I agree, I'm just not sure how stubbing works. When stubbing openPR, can I check arguments for the expected update objects?

Reply:
Okay, I put in a fake for openPR that checks the expected versions.